### ISSUE TITLE
Support for events on proxies

### ISSUE THREAD

**Issue by dpavlenkov** [louthy/echo-process#8]
For transparency, supporting events on proxies would be nice. It's a sort of wrapped pub/sub.

**Comment by louthy** [comment]
Yep, that's a good idea.  I have a similar solution in another project.  Not sure when I'll get to it though, but I'll definitely get around to it at some point.
